crates/ruff_notebook/src/notebook.rs

```diff
@@ -146,15 +146,33 @@ impl Notebook {
         // https://github.com/astral-sh/ruff/issues/6834
         // https://github.com/jupyter/enhancement-proposals/blob/master/62-cell-id/cell-id.md#required-field
         if raw_notebook.nbformat == 4 && raw_notebook.nbformat_minor >= 5 {
+            let mut id_index: u128 = 0;
+            let mut seen_ids = HashSet::new();
```

Comment:
We should collect all of the ids beforehand, otherwise we could generate an id that is present in a later cell.